dbms/src/IO/StdStreamBufFromReadBuffer.h

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <memory>
+#include <streambuf>
+
+
+namespace DB
+{
+class ReadBuffer;
+class ReadBufferFromFileBase;
+
+/// `std::streambuf`-compatible wrapper around a ReadBuffer.
+/// Warning:
+/// The implementation this class is quite tricky and not complete.
+/// We need to continuously keep the pointers of get area in sync by calling `setg()` and changing `position()`.
+/// Please use it with caution and do enough tests.
```

Comment:
Can we add some ut about the known issue of `StdStreamBufFromReadBuffer` and `StdStreamFromReadBuffer`?

Reply:
currently no known issue, I have added some ut about common usage of reading from std::stream;